src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -213,6 +228,9 @@ def create_frames_from_structure(
         Indicates if structure files are gzipped or not.
     verbosity: int
         Level of logging sent to std out.
+    encode_cb: bool,
+        Whether to encode the Cb at a separate channel of the frame. If
```

Comment:
Good catch, that was actually a mistake!

My overall plan at the moment is for the extra Cb channel was to do it in a posteriori (probably somewhere in the generator).

In this case, `encode_cb` makes sure the avg Cb coordinates are added to the frame.

Reply:
Hmm, not sure you're going to be able to do that in `aposteriori`, as we'll loose the [residue label](https://github.com/isambard-uob/ampal/blob/906e2afacb435ffb129b381f262ff8e7bfb324c5/src/ampal/base_ampal.py#L644) information as we only store the atomic number. We could store it in the attrs, but it'd need to be a list with the label for every atom in the frame, and we'd need some way to resolve that. We could calculate it from connectivity, but it would be slow, prone to error and make the base dataset unrepresentative of the model input (which I want to avoid now). I think that the most pragmatic solution is to have it as a flag for dataset creation and then sticking encoding information in the top level attrs of the dataset _i.e._:

```python
hd5.attrs['input_label'] = ['N', 'CA', 'C', 'O', 'CB,]
hd5.attrs['int_representation'] = [0, 1, 2, 3, 4]
```
It'd be nice if the dataset was completely self contained. Let me know what you think. Obviously, you don't need to implement this encoding just now if you're wanting to train models with the element based encoding, but it'd be worth adding a feature issue for it.

Reply:
Uh, I was thinking about a much simpler, perhaps hacky way. 

The Cb will always be at the same position either way (even for glycine). A simple (perhaps not elegant) if statement could check whether a C is present at `(-0.741287356, - 0.53937931, - 1.224287356)` and given an option eg `encode_cb_as_separate_channel` it could use a different number (say "5") and use a different one-hot-encoder. 

This point needs to be thought over a bit more deeply. Should we open a separate issue?

Reply:
I'm not sure that I'm understanding you, remember we'd want all the CB vectors to be labelled in the frame.